test/e2e/commands/node-update.test.mjs

```diff
@@ -0,0 +1,101 @@
+  const defaultTimeout = 120000
+  const namespace = 'node-update'
+  const nodeId = 'node3'
+  const argv = getDefaultArgv()
+  argv[flags.keyFormat.name] = constants.KEY_FORMAT_PEM
+  argv[flags.nodeIDs.name] = 'node1,node2,node3'
+  argv[flags.nodeID.name] = nodeId
+
+  argv[flags.newAccountNumber.name] = '0.0.7'
+  argv[flags.newGRPCHash.name] = path.join('test', 'data', 'new_tls.crt')
+  argv[flags.newCACertificate.name] = path.join('test', 'data', 'new_sign.pem')
+  argv[flags.newAdminKey.name] = '302e020100300506032b6570042204200cde8d512569610f184b8b399e91e46899805c6171f7c2b8666d2a417bcc66c2'
+
+  argv[flags.generateGossipKeys.name] = true
+  argv[flags.generateTlsKeys.name] = true
+  argv[flags.keyFormat.name] = constants.KEY_FORMAT_PEM
+  // set the env variable SOLO_FST_CHARTS_DIR if developer wants to use local FST charts
+  argv[flags.chartDirectory.name] = process.env.SOLO_FST_CHARTS_DIR ? process.env.SOLO_FST_CHARTS_DIR : undefined
+  argv[flags.releaseTag.name] = HEDERA_PLATFORM_VERSION_TAG
+  argv[flags.namespace.name] = namespace
+  argv[flags.persistentVolumeClaims.name] = true
+  const bootstrapResp = bootstrapNetwork(namespace, argv)
+  const nodeCmd = bootstrapResp.cmd.nodeCmd
+  const accountCmd = bootstrapResp.cmd.accountCmd
+  const k8 = bootstrapResp.opts.k8
+  let existingServiceMap
+  let existingNodeIdsPrivateKeysHash
+
+  afterAll(async () => {
+    await getNodeLogs(k8, namespace)
+    await k8.deleteNamespace(namespace)
+  }, 600000)
+
+  it('cache current version of private keys', async () => {
+    existingServiceMap = await nodeCmd.accountManager.getNodeServiceMap(namespace)
+    existingNodeIdsPrivateKeysHash = await getNodeIdsPrivateKeysHash(existingServiceMap, namespace, constants.KEY_FORMAT_PEM, k8, getTmpDir())
+  }, defaultTimeout)
+
+  it('should succeed with init command', async () => {
+    const status = await accountCmd.init(argv)
+    expect(status).toBeTruthy()
+  }, 450000)
+
+  it('should update a new node property successfully', async () => {
+    await nodeCmd.update(argv)
+    expect(nodeCmd.getUnusedConfigs(NodeCommand.UPDATE_CONFIGS_NAME)).toEqual([
+      flags.app.constName,
+      flags.devMode.constName
+    ])
+    await nodeCmd.accountManager.close()
+  }, 600000)
+
+  balanceQueryShouldSucceed(nodeCmd.accountManager, nodeCmd, namespace)
+
+  accountCreationShouldSucceed(nodeCmd.accountManager, nodeCmd, namespace)
+
+  it('existing nodes private keys should not have changed', async () => {
+    const currentNodeIdsPrivateKeysHash = await getNodeIdsPrivateKeysHash(existingServiceMap, namespace, constants.KEY_FORMAT_PEM, k8, getTmpDir())
+
+    for (const [nodeId, existingKeyHashMap] of existingNodeIdsPrivateKeysHash.entries()) {
+      const currentNodeKeyHashMap = currentNodeIdsPrivateKeysHash.get(nodeId)
+
+      for (const [keyFileName, existingKeyHash] of existingKeyHashMap.entries()) {
+        expect(`${nodeId}:${keyFileName}:${currentNodeKeyHashMap.get(keyFileName)}`).toEqual(
```

Comment:
shouldn't they not match for node3?  It looks like you updated the gossip and grpcTLS keys.

Reply:
should hedera app generate new keys after handling nodeUpdateTransaction
it seems only regenerated public key

```
ls -ltr /opt/hgcapp/services-hedera/HapiApp2.0/data/upgrade/current/
total 32
drwxr-xr-x 3 hedera hedera 4096 Aug 28 03:37 data
-rwxr-xr-x 1 hedera hedera 1464 Aug 28 03:37 s-public-node3.pem
-rwxr-xr-x 1 hedera hedera 1464 Aug 28 03:37 s-public-node2.pem
-rwxr-xr-x 1 hedera hedera 1464 Aug 28 03:37 s-public-node1.pem
-rwxr-xr-x 1 hedera hedera    3 Aug 28 03:37 execute_immediate.mf
-rwxr-xr-x 1 hedera hedera  484 Aug 28 03:37 config.txt
-rwxr-xr-x 1 hedera hedera   10 Aug 28 03:37 freeze_scheduled.mf
-rwxr-xr-x 1 hedera hedera    3 Aug 28 03:37 now_frozen.mf
```

Reply:
I don't think it is generating new keys.  There is a download task that will copy down the public keys from nodeId[0] from data/upgrade/current along with the config.txt, then update the configmap, which you have to killl/start all the pods to pick up the change.

Reply:
The operation of download files from dat/upgrade/current and update the configmap are already there.
but the helper function getNodeIdsPrivateKeysHash only get hashes of private key files,
since private key files are untouched, of course the check would pass.

We could manually copy the test key pem file to the key directory, of course it will guarantee the mismatch
but seems an unnecessary move, 

maybe we could just skip the check step.

Reply:
the private and public keys for the TLS and the signing gossip should be updated, so it should not match.  I made notes in other places, as to why I think the code that uploads them are missing.

I'm actually wondering if there is a bug as to why services and platform is working with node2. having the old keys.  or perhaps it is throwing an error and we just aren't looking at the logs?